W-13796916 loading error (browser extension) (#97)
mulesoft/docs-code-builder: modules/ROOT/pages/troubleshoot-app-loading-errors.adoc

= Application Loading Errors

include::reuse::partial$beta-banner.adoc[tag="anypoint-code-builder"]

//LOGO (web, desktop, or both)
include::partial$acb-ide-logos.adoc[tags="both-ides"]

Errors sometimes occur when you open an application in Anypoint Code Builder. 

> == Loading Error in the UI Canvas

Author: @pablocabrera85 Can you clarify your comment here:

> This does not only apply to the UI canvas, but also to the create project UIs

Author: Does the Loading... error occur, or what happens? 

I'm not clear what the difference between "project UIs" and "UI canvas" is. Do you mean some other parts of the open project might not load? Can you show me the error?

Author: Changed to "Loading Error in the Application UI"

> A *Loading...* error in the UI canvas of an open application typically indicates that a browser extension is interfering with the UI loading process. 

Author: ```suggestion
A *Loading...* error in the UI of an open application typically indicates that a browser extension is interfering with the UI loading process. 
```

To address this issue, try:

* Disabling browser extensions and reloading the application 
* Using another browser that is not using extensions

> == Flow Canvas Preparation Never Completes

Author: ```suggestion
== Canvas UI Renders Without Configured Flows
```

> == Canvas UI Renders Without Configured Flows

Author: ```suggestion
== Canvas UI Renders Without Configured Flows or Subflows
```

> If the UI canvas contains the message *Preparing Flow Canvas* but the flow graphics fail to load after a few minutes, make sure that your application is open from its _root_ directory or from a multi-root workspace. 

Author: ```suggestion
If the UI canvas for an application that is already configured with a flow or subflow treats your application as a new project, prompting you with the question *What do you want to build in this file?* and *Flow* and *Sub flow* creation options, make sure that your application is open from its _root_ directory or from a multi-root workspace. 
```

Visual Studio Code treats a project directory as a workspace when you open it to the project's root directory or if your project folder is located in a https://code.visualstudio.com/docs/editor/multi-root-workspaces[multi-root workspace^].

> . Open the Command Palette by pressing Ctrl+Shift+p (Windows) or Cmd+Shift+p (Mac), and provide the following command:

Reviewer: ```suggestion
. Open the Command Palette by pressing `Ctrl+Shift+p` (Windows) or `Cmd+Shift+p` (Mac), and provide the following command:
```

Author: @dmerlob Apparently, the Salesforce style does not use a different font face for the key command shortcuts. I don't know why, and it's unintuitive to me, but it is what it is. ¯\_(ツ)_/¯ Hanna pointed me to that guidance.

= Application Loading Errors

include::reuse::partial$beta-banner.adoc[tag="anypoint-code-builder"]

//LOGO (web, desktop, or both)
include::partial$acb-ide-logos.adoc[tags="both-ides"]

Errors sometimes occur when you open an application in Anypoint Code Builder. 

== Loading Error in the Application UI

A *Loading...* error in the UI of an open application typically indicates that a browser extension is interfering with the UI loading process. 

image::troubleshoot-loading-error.png["Loading Error in the UI"]

To address this issue, try:

* Disabling browser extensions and reloading the application 
* Using another browser that is not using extensions

== Canvas UI Renders Without Configured Flows or Subflows

If the UI for an application that is already configured with a flow or subflow treats your application as a new project, prompting you with the question *What do you want to build in this file?* and *Flow* and *Sub flow* creation options, make sure that your application is open from its _root_ directory or from a multi-root workspace. 

image::acb-start-flow-subflow-ui.png["Start Screen Building a Mule Application"]

> Visual Studio Code treats a project directory as a workspace when you open it to the project's root directory or if your project folder is located in a https://code.visualstudio.com/docs/editor/multi-root-workspaces[multi-root workspace^].

Reviewer 2: I find this a bit hard to follow. What do you think of something around `Visual Studio Code treats an open directory as a workspace. Even if this folder contains several projects, they won't be treated as such. In order to have multiple projects open in the same editor, you need to create a https://code.visualstudio.com/docs/editor/multi-root-workspaces[multi-root workspace^].`

Author: ok, I think I see what you mean. thx

Author: ```suggestion
Visual Studio Code treats an open directory as a workspace. For the project to load into a workspace, the root directory of the project must be the top-level directory in the Explorer view, or the project must belong to a https://code.visualstudio.com/docs/editor/multi-root-workspaces[multi-root workspace^].
```

> Visual Studio Code treats an open directory as a workspace. For the project to load into a workspace, the root directory of the project must be the top-level directory in the Explorer view, or the project must belong to a https://code.visualstudio.com/docs/editor/multi-root-workspaces[multi-root workspace^].

Author: revised @pablocabrera85 
can you take a look
